app/models/project_media.rb

```diff
@@ -475,21 +475,95 @@ def get_similar_articles
       search_query ||= self.title
       results = self.team.search_for_similar_articles(search_query, self)
       fact_check_ids = results.select{|article| article.is_a?(FactCheck)}.map(&:id)
+      fc_pm = {}
+      unless fact_check_ids.blank?
+        # Get ProjectMedia for FactCheck for RelevantResultsItem logs and should use sort_by to keep existing order
+        FactCheck.select('fact_checks.id AS fc_id, claim_descriptions.project_media_id AS pm_id')
+        .where(id: fact_check_ids).joins(:claim_description).each do |raw|
+          fc_pm[raw.fc_id] = raw.pm_id
+        end
+      end
       explainer_ids = results.select{|article| article.is_a?(Explainer)}.map(&:id)
-      { fact_check: fact_check_ids, explainer: explainer_ids }.to_json
+      ex_pm = {}
+      unless explainer_ids.blank?
+        # Intiate the ex_pm with nil values as some Explainer not assinged to existing items
+        default_pm = nil
+        ex_pm = explainer_ids.each_with_object(default_pm).to_h
+        # Get ProjectMedia for Explainer for RelevantResultsItem logs and should use sort_by to keep existing order
+        ExplainerItem.where(explainer_id: explainer_ids).find_each do |raw|
+          ex_pm[raw.explainer_id] = raw.project_media_id
+        end
+      end
```

Comment:
Do you really need to store all `project_media_id`'s? I was expecting that the `matched_media_id` was actually `self` (so, the input `ProjectMedia`) for all cases.

Reply:
yes Caio, Based on [shared doc](https://docs.google.com/spreadsheets/d/1ZqpI_twMF2D_JFal_8o3UJXQeWu0fjZjYe9bTOXAfDM/edit?gid=283126175#gid=283126175)
the `matched_media_id` is project media that was returned from similarity search (which an Article was linked to)
and `query_media_parent_id`  is parent id of cluster used

Reply:
OK, then!